packages/internal/src/generate/graphqlCodeGen.ts

```diff
@@ -133,23 +134,63 @@ function getLoadDocumentsOptions(filename: string) {
   return loadTypedefsConfig
 }
 
+function getPrismaClient(hasGenerated = false): {
+  ModelName: Record<string, string> | undefined
+} {
+  const localPrisma = require('@prisma/client')
+
+  if (!localPrisma.ModelName) {
+    if (hasGenerated) {
+      return { ModelName: undefined }
+    } else {
+      if (process.env.NODE_ENV === 'test') {
+        // When running as part of tests `yarn rw` isn't available as a bin
+        // In actual projects it will be available
+        return { ModelName: undefined }
+      }
+
+      execa.sync('yarn rw prisma generate', { shell: true })
+
+      // Purge Prisma Client from node's require cache, so that the newly
+      // generated client gets picked up by any script that uses it
+      Object.keys(require.cache).forEach((key) => {
+        if (
+          key.includes('/node_modules/@prisma/client/') ||
+          key.includes('/node_modules/.prisma/client/')
+        ) {
+          delete require.cache[key]
+        }
+      })
+
+      return getPrismaClient(true)
+    }
+  }
+
+  return localPrisma
+}
+
 function getPluginConfig() {
-  let prismaModels: Record<string, string> = {}
-  try {
-    // Extract the models from the prisma client and use those to
-    // set up internal redirects for the return values in resolvers.
-    const localPrisma = require('@prisma/client')
-    prismaModels = localPrisma.ModelName
+  // Extract the models from the prisma client and use those to
+  // set up internal redirects for the return values in resolvers.
+  const localPrisma = getPrismaClient()
+  const prismaModels = localPrisma.ModelName
+
+  if (prismaModels) {
```

Comment:
Would it be a lot less code here to have an empty object instead of having `ModelName` as undefined?

Reply:
It was one of the first things I tried. But something broke. However, some of the other changes I've made later must have fixed whatever it was that made it brake, because now your suggestion works, and looks so much cleaner! Thanks for making me try it again.